channel.go

```diff
@@ -1476,17 +1480,16 @@ exception could occur if the server does not support this method.
 
 */
 func (me *Channel) Confirm(noWait bool) error {
-	me.m.Lock()
-	defer me.m.Unlock()
-
 	if err := me.call(
 		&confirmSelect{Nowait: noWait},
 		&confirmSelectOk{},
 	); err != nil {
 		return err
 	}
 
+	me.m.Lock()
 	me.confirming = true
```

Comment:
Bet advised - the `me.confirming` is also accessed:
https://github.com/streadway/amqp/blob/master/channel.go#L276
https://github.com/streadway/amqp/blob/master/channel.go#L285

Without any sort of mutex.